integration/stability/bind_mount_linux.sh

```diff
@@ -0,0 +1,64 @@
+#!/bin/bash
+#
+# Copyright (c) 2018 Ning Lu
+#
+# SPDX-License-Identifier: Apache-2.0
+#
+# This test will start a container with a bind mount
+# and set bind propagation, the purpose of this
+# test is to check if the container stops properly
+
+set -e
+
+cidir=$(dirname "$0")
+testname="${0##*/}"
+sysname=$(uname)
+
+[ "$sysname"x == "Linux"x ] || exit 0
+
+source "${cidir}/../../lib/common.bash"
+
+# Environment variables
+IMAGE="${IMAGE:-busybox}"
+CONTAINER_NAME="${CONTAINER_NAME:-test}"
+PAYLOAD_ARGS="${PAYLOAD_ARGS:-tail -f /dev/null}"
+MOUNT_DIR="/mnt/tmp"
+BIND_DIR="${MOUNT_DIR}/sub"
+DOCKER_ARGS="-v ${MOUNT_DIR}:${MOUNT_DIR}:rslave"
+CONTAINER_ID=
+
+# Set the runtime if not set already
+RUNTIME="${RUNTIME:-kata-runtime}"
+
+function setup {
+	clean_env
+	docker run --runtime=${RUNTIME} -d ${DOCKER_ARGS} --name ${CONTAINER_NAME} ${IMAGE} ${PAYLOAD_ARGS}
+	CONTAINER_ID=$(docker ps -q -f "name=${CONTAINER_NAME}")
+}
+
+function cmd_bind_mount {
+	mkdir -p ${BIND_DIR}
+	mount --bind /tmp ${BIND_DIR}
+	docker rm -f ${CONTAINER_NAME}
+}
+
+function check {
+	kata_pid_count=$(ps aux | grep ${CONTAINER_ID} | grep -v grep | wc -l)
+	if (( kata_pid_count != 0 ));then
+		die "Left $kata_pid_count kata processes, quitting"
```

Comment:
How about displaying full details of these "rogue" processes before dying? That might help with debugging any issues. Also, should those processes continue to run as this is a CI test? Maybe they should be killed and the bind mount unmounted?

/cc @chavafg, @grahamwhaley.

Reply:
ok, update to clean processes and mount point